File: 3-arrays/4-common-array-problems/main.cpp
// Common Array Problems - C++ Example

#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    std::vector<int> arr = {1, 3, 2, 8, 5};

    // Find maximum
    int max_val = *std::max_element(arr.begin(), arr.end());
    std::cout << "Maximum value: " << max_val << std::endl;

    // Reverse array
    std::reverse(arr.begin(), arr.end());
    std::cout << "Reversed array: ";
    for (int val : arr) std::cout << val << " ";
    std::cout << std::endl;

    // Sum of subarray [1, 4)
    int sub_sum = 0;
    for (int i = 1; i < 4; i++) {
        sub_sum += arr[i];
    }
    std::cout << "Subarray sum: " << sub_sum << std::endl;

    return 0;
}

File: 3-arrays/4-common-array-problems/main.java
// Common Array Problems - Java Example

import java.util.Arrays;

public class CommonArrayProblems {

    public static void main(String[] args) {
        int[] arr = {1, 3, 2, 8, 5};

        // Find maximum
        int maxVal = Arrays.stream(arr).max().getAsInt();
        System.out.println("Maximum value: " + maxVal);

        // Reverse array
        for (int i = 0; i < arr.length / 2; i++) {
            int temp = arr[i];
            arr[i] = arr[arr.length - 1 - i];
            arr[arr.length - 1 - i] = temp;
        }
        System.out.print("Reversed array: ");
        for (int val : arr) {
            System.out.print(val + " ");
        }
        System.out.println();

        // Sum of subarray [1, 4)
        int subSum = 0;
        for (int i = 1; i < 4; i++) {
            subSum += arr[i];
        }
        System.out.println("Subarray sum: " + subSum);
    }
}
